raiden/network/transport/matrix/transport.py

```diff
@@ -953,79 +964,75 @@ def _get_room_for_address(
         address_hex = to_normalized_address(address)
         msg = f"address not health checked: me: {self._user_id}, peer: {address_hex}"
         assert address and self._address_mgr.is_address_known(address), msg
+        assert self._raiden_service is not None
 
-        # filter_private is done in _get_room_ids_for_address
-        room_ids = self._get_room_ids_for_address(address)
-        if room_ids:  # if we know any room for this user, use the first one
-            # This loop is used to ignore any global rooms that may have 'polluted' the
-            # user's room cache due to bug #3765
-            # Can be removed after the next upgrade that switches to a new TokenNetworkRegistry
-            while room_ids:
-                room_id = room_ids.pop(0)
-                room = self._client.rooms[room_id]
-                if not self._is_room_global(room):
-                    self.log.debug("Existing room", room=room, members=room.get_joined_members())
-                    return room
-                self.log.warning("Ignoring global room for peer", room=room, peer=address_hex)
-
-        assert self._raiden_service is not None, "_raiden_service not set"
-        address_pair = sorted(
-            [to_normalized_address(address) for address in [address, self._raiden_service.address]]
-        )
-        room_name = make_room_alias(self.chain_id, *address_pair)
-
-        # no room with expected name => create one and invite peer
+        # filter peer_candidates
         peer_candidates = [
             self._get_user(user) for user in self._client.search_user_directory(address_hex)
         ]
-
-        # filter peer_candidates
         peers = [user for user in peer_candidates if validate_userid_signature(user) == address]
+
         if not peers and not allow_missing_peers:
             self.log.error("No valid peer found", peer_address=to_checksum_address(address))
             return None
 
+        # filter_private is done in _get_room_ids_for_address
+        room_ids = self._get_room_ids_for_address(address)
+        online_user_ids = self._address_mgr.get_online_user_ids_for_address(address)
+        while room_ids:
+            room = self._client.rooms[room_ids.pop()]
+            member_ids = {member.user_id for member in room.get_joined_members()}
+
+            if member_ids & online_user_ids:
```

Comment:
The client verifies that the partner is online and otherwise rejects the route. 
What we do here is check that _some_ user_id of the partner is online, which is also important for roaming. If
address_a has user_id1, 2, 3 at transport1, 2 and 3 we might otherwise just return a room with user_id1 (offline), while user user_id 3 is online right now. 

Also this no hard requirement, as the function will return empty/non_online rooms if no match was found here.

Reply:
I believe this condition here is incorrect. It is perfectly fine for a node to go offline, and if that is the case I don't think a new room should be created. (A new room would be created because the loop would exit and the next thing this function does is create a new room).